### PR TITLE
docs(dxp): add code review notes

### PR DESCRIPTION
**Pull request by wincent**
These are specifically for the Frontend Infrastructure Team, but may be of general interest; in any case, each team is free to develop its own processes.

Consider this a work-in-progress, as the liferay-frontend-based workflow is something we are newly experimenting with.

### CODE REVIEW

File: dxp/code_review.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+3. **Always provide a description.** In order to expedite review, give the reviewer(s) all the context they need to understand what your PR aims to do, and any analysis that you undertook to arrive at your solution (this can include a description of alternatives that you _didn't_ select, and why) (see [example](https://github.com/liferay-frontend/liferay-portal/pull/2)).
+4. **Always provide a test plan.** This is part of the description, and consists of the steps you took to verify that your change is correct (see [example](https://github.com/liferay-frontend/liferay-portal/pull/34)). This serves two purposes: one is to enable the reviewer to easily check your PR (if they deem it necessary); the other is to provide the reviewer with information that may save them the effort of testing it themselves (that is, if your test plan is good, the reviewer can feel more confident about the change).
+5. **Always assign a reviewer.** All team members are collaborators on the repo and may be assigned as reviewers. Proactively selecting reviewers and `@`-mentioning them will minimize the amount of time your full sits unattended in the queue.
```
Comment (jbalsas) [inline dxp/code_review.md:9]: your **f**ull maybe means your **p**ull?

**Comment by jbalsas** [comment]
Looks great to me! 👍 

Only thing a bit unclear to me is the target audience... at times it feels directed specifically to members of the Frontend Infra team and at times to people in general... no big deal, though.

**Comment by wincent** [comment]
> Looks great to me! 👍
> 
> Only thing a bit unclear to me is the target audience... at times it feels directed specifically to members of the Frontend Infra team and at times to people in general... no big deal, though.

Yeah, I was specifically going for the team (even though I know this repo isn't just for the team). When I was writing it up, I actually had an introductory sentence or two explaining that — something along the lines of that it was guidelines for our practices but we're sharing it because it may be helpful for other teams too as they develop their own processes — so I might add it back.

**Comment by jbalsas** [comment]
Merging, so we have this in stone... no turning back now!!

moses

**Pull request merged**